llvm/lib/Transforms/InstCombine/InstCombineMulDivRem.cpp

```diff
@@ -1820,6 +1989,30 @@ Instruction *InstCombinerImpl::visitFDiv(BinaryOperator &I) {
     return R;
 
   Value *Op0 = I.getOperand(0), *Op1 = I.getOperand(1);
+
+  // Convert
+  // x = 1.0/sqrt(a)
+  // r1 = x * x;
+  // r2 = a * x;
+  //
+  // TO
+  //
+  // tmp1 = 1.0 / a
+  // tmp2 = sqrt (a)
+  // tmp3 = tmp1 * tmp2
+  // x = tmp3
+  // r1 = tmp1
+  // r2 = tmp2
+  SmallSetVector<Instruction *, 2> R1, R2;
+  getFSqrtDivOptPattern(&I, R1, R2);
+  if (R1.size() && R2.size() &&
+      isFSqrtDivToFMulLegal(static_cast<Instruction *>(&I), R1, R2)) {
+    CallInst *CI = static_cast<CallInst *>((&I)->getOperand(1));
```

Comment:
Don't use static_cast, use cast

Reply:
do you mean C style casts?       e.g. in above case,   (CallInst*)(....)      ?